app/src/main/java/de/westnordost/streetcomplete/tangram/QuestsMapFragment.java

```diff
@@ -268,10 +281,18 @@ protected void updateView()
 		retrievedTiles.addAll(tiles);
 	}
 
-	public void setQuestYOffsets(int questTopOffset, int questBottomOffset)
+	public void setQuestOffsets()
 	{
-		this.questTopOffset = questTopOffset;
-		this.questBottomOffset = questBottomOffset;
+		questOffset = new Rect();
+		questOffset.top = 50;
+		if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_PORTRAIT)
+		{
+			questOffset.bottom = getResources().getDimensionPixelSize(R.dimen.quest_bottom_sheet_peek_height);
+		}
+		else if (getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE)
+		{
+			questOffset.left = getResources().getDimensionPixelSize(R.dimen.quest_bottom_sheet_landscape_width);
+		}
```

Comment:
Same here. How the quest bottom sheet is layouted, is managed by the `MainActivity`, not `QuestsMapFragment`. So basically, you are creating an indirect interdependency between these two classes here.

Generally, think twice when you think you need to add a configuration (orientation) check in code. It is a code smell that hints to bad design - configurations should be defined in the resources xml.

Here, it  should be `setQuestsOffsets(Rect offsets)` and the MainActivity passes in the rect. And the MainActivity constructs the rect like this (pseudo code):
```
rect = new Rect(R.dimen.quest_form_leftOffset, R.dimen.quest_form_topOffset, R.dimen.quest_form_rightOffset ,R.dimen.quest_form_bottomOffset)
```

and then define the appropriate values in the `dimens.xml` for the different configurations. I.e.
`left=0, right=0, top=height of top bar, bottom=peek height for bottom sheet` in portrait mode (normal) and `left=bottom sheet width, right=0, top=height of top bar, bottom=0` for landscape mode.